tests/load-tests.php

```diff
@@ -97,6 +97,32 @@ function( $default, $option, $passed_default ) use ( &$has_passed_default ) {
 		$this->assertSame( $new_value, $settings );
 	}
 
+	/**
+	 * @dataProvider data_legacy_modules
+	 */
+	public function test_legacy_module_for_perflab_get_module_settings( $legacy_module_slug, $current_module_slug ) {
+		$new_value = array(
+			'site-health/audit-autoloaded-options' => array( 'enabled' => true ),
+			'site-health/audit-enqueued-assets'    => array( 'enabled' => true ),
+			'site-health/audit-full-page-cache'    => array( 'enabled' => true ),
+			'site-health/webp-support'             => array( 'enabled' => true ),
+		);
+		update_option( PERFLAB_MODULES_SETTING, $new_value );
+
+		$settings = perflab_get_module_settings();
+		$this->assertArrayNotHasKey( $legacy_module_slug, $settings, 'The settings do not contain the old legacy module slug in the database' );
+		$this->assertArrayHasKey( $current_module_slug, $settings, 'The settings contain an updated module slug in the database' );
+	}
+
+	public function data_legacy_modules() {
```

Comment:
@mukeshpanchal27 can you add function description here like `Data provider for legacy modules` and what the function returns via `@return array`?

Reply:
Thanks for the feedback! Docblock added for the `data_legacy_modules()`